### PR TITLE
45 new variable statistics final energy by carriernatural gas

### LINKED ISSUE #45: New Variable Statistics: Final Energy [by Carrier]|Natural Gas
### Goal: 
**Add a new function to extract the statistics of variable Final Energy [by Carrier]|Natural Gas**

> **Definition:** Energy consumption of fossil methane excluding transmission/distribution losses

### To Dos:
- [x] Create a new branch linked to this issue
- [ ] Write your pypsa-statistics and add it as a separate function to [statistics_functions.py](https://github.com/maxnutz/pypsa_validation_processing/blob/main/pypsa_validation_processing/statistics_functions.py) (please note the [naming and structural conventions](https://github.com/maxnutz/pypsa_validation_processing/tree/main#variables-statistics---functions)!)
- [ ] add a comprehensive docstring to your function 
- [ ] add the mapping variable_name <> function_name to [mapping.default.yaml](https://github.com/maxnutz/pypsa_validation_processing/blob/main/pypsa_validation_processing/configs/config.default.yaml) (and your personal mapping-file)
- [ ] Add a testing routine for your Function to `tests/` - stick to the [testing-README](https://github.com/maxnutz/pypsa_validation_processing/blob/main/tests/README.md)
- [ ] make sure, that the newest version of main is merged into your feature Branch 
- [ ] open a pull request and assign @maxnutz as reviewer

### PR DESCRIPTION
**Pull request by maxnutz**
## Goal
Statistics-function for variable `Final Energy [by Carrier]|Natural Gas` 

> **Definition:** Energy consumption of fossil methane excluding transmission/distribution losses

### To Dos
- [x] add a comprehensive docstring to your function 
- [x] add the mapping variable_name <> function_name to [mapping.default.yaml](https://github.com/maxnutz/pypsa_validation_processing/blob/main/pypsa_validation_processing/configs/config.default.yaml) (and your personal mapping-file)
- [x] Add a testing routine for your Function to `tests/` - stick to the [testing-README](https://github.com/maxnutz/pypsa_validation_processing/blob/main/tests/README.md)
- [x] make sure, that the newest version of main is merged into your feature Branch 

## Validation

| variable | PyPSA-output | IAMC Energy Balance | 🟢 🟡 🟠 🔴 |
| -- | -- | -- | -- |
| Final Energy [by Carrier\|Natural Gas | 221754 | 196259 | 🟠 too high |

- carrier `gas for industry` includes non-energetic use of gas in industry-sector. The energetic use of natural gas is estimated by a general fraction for EU27. 

## Summary by Sourcery

Add a statistics function to compute final energy consumption from fossil natural gas and integrate it into the variable mapping and unit handling.

New Features:
- Introduce Final_Energy_by_Carrier__Natural_Gas statistics function to calculate final energy consumption from fossil natural gas across sectors.

Enhancements:
- Extend unit mapping to explicitly support MWh unit strings in statistics results.

Tests:
- Add a dedicated test suite covering typical and edge-case scenarios for the Final_Energy_by_Carrier__Natural_Gas statistics function.

## Summary by Sourcery

Add a statistics function to compute final energy consumption from natural gas, integrate it into the processing pipeline, and strengthen supporting utilities and tests.

New Features:
- Introduce Final_Energy_by_Carrier__Natural_Gas statistics function to calculate fossil natural gas final energy demand across residential, commercial, and industry sectors.

Enhancements:
- Ensure statistics functions that do not aggregate per year return DataFrames, raising an error otherwise.
- Extend unit mapping to explicitly support plain MWh units in statistics results.
- Adjust the workflow_test pixi task definition formatting for improved multi-command execution.

Tests:
- Add a dedicated test suite covering typical and edge-case scenarios for the Final_Energy_by_Carrier__Natural_Gas statistics function.

### CODE REVIEW

**Comment by maxnutz** [comment]
@sourcery-ai review

File: pypsa_validation_processing/statistics_functions.py [hunk 1]
```diff
@@ -133,6 +134,125 @@ def Final_Energy_by_Carrier__Electricity(
     return result
 
 
+def Final_Energy_by_Carrier__Natural_Gas(
+    n: pypsa.Network,
+    aggregate_per_year: bool = True,
```
[the remark this answers is not in the document]
Comment (maxnutz) [reply pypsa_validation_processing/statistics_functions.py:314]: outdated

File: tests/test_statistics_functions.py [hunk 2]
```diff
@@ -219,6 +220,181 @@ def test_issues_expected_withdrawal_queries(self):
...
+
+    def _natural_gas_network(self, scenario: str = "mixed"):
+        return self._NaturalGasNetwork(scenario=scenario)
+
+    def test_returns_series(self):
+        """Function returns a Series with expected output format."""
+        result = Final_Energy_by_Carrier__Natural_Gas(self._natural_gas_network())
+        assert isinstance(result, pd.Series)
+        assert isinstance(result.index, pd.MultiIndex)
+        assert result.index.names == ["location", "unit"]
```
[the remark this answers is not in the document]
Comment (maxnutz) [reply tests/test_statistics_functions.py:673]: implemented!
